on-target-tests/tests/multicore.rs

```diff
@@ -100,18 +100,46 @@ mod tests {
         let counter = super::COUNTER.load(Ordering::Acquire);
         assert_eq!(2 * super::STEPS, counter);
     }
+
+    #[test]
+    fn check_floats() {
+        super::STATE.store(3, Ordering::Release);
+        super::calculations();
+        while super::STATE.load(Ordering::Acquire) != 4 {}
+
+        let counter = super::COUNTER.load(Ordering::Acquire);
+        assert_eq!(2 * super::STEPS, counter);
```

Comment:
Is this right? COUNTER only changes when STATE is 1? Perhaps it could do 1.5 * 2.0 or something and store the result as an integer to be checked on Core 0?

Reply:
No, it's not. I forgot to remove these lines and the test succeeds accidentally because the other test case runs first and leaves the counter at that value.

It's not needed to have an assert here. If FP Code doesn't work properly on core1, it will fault and STATE won't be set to 4. It's a bit unfortunate that this just hangs the test instead of returning a useful error message, but if the test passes it shows that the FPU has been properly initialized.

BTW the double-precision coprocessor doesn't work at all in the test cases because they don't use the entry macro.